src/NUnitFramework/framework/Constraints/CollectionEquivalentConstraintResult.cs

```diff
@@ -0,0 +1,113 @@
+    /// </summary>
+    /// <remarks>
+    /// Assumes that a <see cref="CollectionTally"/> has been created and used to compare the expected
+    /// equivalent collections.
+    /// </remarks>
+    public class CollectionEquivalentConstraintResult : ConstraintResult
+    {
+        /// <summary>
+        /// Preprocessed <see cref="CollectionTally"/> of the collections to compare for equivalence.
+        /// </summary>
+        private CollectionTally _tally;
+
+        /// <summary>
+        /// Maximum amount of elements to write to the <see cref="MessageWriter"/> if there are
+        /// extra/missing elements from the collection.
+        /// </summary>
+        private const int MaxDifferingElemsToWrite = 10;
+        
+        /// <summary>
+        /// Construct a <see cref="CollectionEquivalentConstraintResult"/> with a <see cref="CollectionTally"/>
+        /// already applied to the actual collection.
+        /// </summary>
+        /// <param name="Constraint">
+        /// Source <see cref="CollectionEquivalentConstraint"/>.
+        /// </param>
+        /// <param name="Tally">
+        /// Preprocessed <see cref="CollectionTally"/> for the collections to compare.
+        /// </param>
+        /// <param name="actual">
+        /// Collection to compare to the expected one.
+        /// </param>
+        /// <param name="isSuccess">
+        /// Whether or not the <see cref="CollectionEquivalentConstraint"/> succeeded.
+        /// </param>
+        /// <remarks>
+        /// The already processed <see cref="CollectionTally"/> is provided to reduce
+        /// unnecessary iterations over the collections to determine equivalence.
+        /// </remarks>
+        public CollectionEquivalentConstraintResult(
+            CollectionEquivalentConstraint Constraint,
+            CollectionTally Tally,
+            object actual,
+            bool isSuccess)
+            : base(Constraint, actual, isSuccess)
+        {
+            if (Tally == null)
+            {
+                throw new System.ArgumentNullException("Tally was null.");
+            }
+
+            _tally = Tally;
+        }
+
+        /// <summary>
+        /// Write the custom failure message for this object's 
+        /// <see cref="CollectionEquivalentConstraint"/>.
+        /// </summary>
+        /// <param name="writer"></param>
+        public override void WriteMessageTo(MessageWriter writer)
+        {
+            //Write the expected/actual message first.
+            base.WriteMessageTo(writer);
+            
+            if (_tally.MissingItems.Count > 0)
+            {
+                int missingItemsCount = _tally.MissingItems.Count;
+
+                string missingStr = $"Missing ({missingItemsCount}): ";
+                missingStr += MsgUtils.FormatCollection(_tally.MissingItems, 0, MaxDifferingElemsToWrite);
+
+                writer.WriteMessageLine(missingStr);
+            }
+
+            if (_tally.ExtraItems.Count > 0)
```

Comment:
So if the actual collection is empty, we don't want to output 

> But was:  <empty>

?  I believe one of my tests ensures that line is in the output message, so that would be a change.

If we do this, that would mean that the end user isn't immediately aware that the expected/actual provided collection was empty.  Would that make sense?

The output would look like
> Expected: < empty >
> Extra(1): < "one" >

or

> Expected: < "one" >
> Missing(1): < "one" > 

I think these two cases would make sense.  Thoughts?  Just making sure we're on the same page.